src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java

```diff
@@ -24,19 +24,21 @@
 import org.springframework.data.annotation.Persistent;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
+import java.util.Arrays;
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
 
 @RunWith(SpringJUnit4ClassRunner.class)
 @PropertySource(value = {"classpath:application.properties"})
 public class DocumentDbTemplateIT {
-
     private static final String TEST_ID = "testid";
     private static final String TEST_NOTEXIST_ID = "testid2";
 
     private static final String TEST_DB_NAME = "testdb";
-    private static final Person TEST_PERSON = new Person(TEST_ID, "testfirstname", "testlastname");
+    private static final List<String> hobbies = Arrays.asList("photography", "fishing");
```

Comment:
could you pls try to test with nested object array to find out where's gap to support nested object?

Reply:
I will do a local test first.